app/models/user.rb

```diff
@@ -4,6 +4,11 @@ class User < ApplicationRecord
   validates :name, presence: true, uniqueness: true
   has_secure_password
   
+  # callbacks extentions
+  before_destroy :ensure_not_delete_if_admin_email
```

Comment:
rename method `check_admin_before_update` and similarly other

Reply:
done